### PR TITLE
Allow empty references (fixes #161)

### LINKED ISSUE #161: Sacrebleu 2.0 dies when encountering an empty reference line
Sacrebleu 2.0.0 dies if there is an empty line in the reference. You can reproduce this with this command:

```bash
 $ sacrebleu -t mtedx/test -l el-en --echo ref | sacrebleu -t mtedx/test -l el-en
Traceback (most recent call last):
  File "/home/mattpost/local/bin/sacrebleu", line 8, in <module>
    sys.exit(main())
  File "/home/mattpost/local/lib/python3.8/site-packages/sacrebleu/sacrebleu.py", line 482, in main
    metrics[name] = METRICS[name](**metric_args)
  File "/home/mattpost/local/lib/python3.8/site-packages/sacrebleu/metrics/bleu.py", line 197, in __init__
    self._ref_cache = self._cache_references(references)
  File "/home/mattpost/local/lib/python3.8/site-packages/sacrebleu/metrics/base.py", line 333, in _cache_references
    raise RuntimeError("Empty or `None` reference sentence found.")
RuntimeError: Empty or `None` reference sentence found.
```

### PR DESCRIPTION
**Pull request by mjpost**

### CODE REVIEW

File: sacrebleu/metrics/base.py [hunk 1]
```diff
@@ -325,12 +325,7 @@ def _cache_references(self, references: Sequence[Sequence[str]]) -> List[Any]:
         num_refs = set()
 
         for refs in zip(*references):
-            # remove undefined / empty references
-            # i.e. we have fewer references for this particular sentence
-            lines = [x for x in refs if x is not None and x != ""]
-
-            if len(lines) == 0:
-                raise RuntimeError("Empty or `None` reference sentence found.")
+            lines = [x for x in refs]
```
Comment (ozancaglayan) [inline sacrebleu/metrics/base.py:328]: You should keep the `None` check here to not break the variable number of references feature as discussed in both the related PR and the issue.

**Review by ozancaglayan — CHANGES_REQUESTED** [review]
README needs updating as well. Also, this change makes the use of variable number of references not possible when refs are given through STDIN or file, instead of the API I think as empty lines will now be proper references that are empty so they'll be used as references. I'll test this now.

**Comment by XapaJIaMnu** [comment]
Is there anything missing for this PR other than the documentation changes? Would you rather have an approach that makes variable number of references continue to work in the API?

**Comment by ozancaglayan** [comment]
Hi,

The discussion in #161 got no replies and stalled. Let me look again into it, to refresh my mind.

**Comment by mjpost** [comment]
Okay, I plan to release v2.2 on Friday, and am going to include this.

File: sacrebleu/metrics/base.py (hunk 1 above)
Comment (mjpost) [reply sacrebleu/metrics/base.py:328]: Done, thanks for the catch!

**Comment by mjpost** [comment]
Merged with a73315b839dee92c9237c30f64380aa36dda4795.

**Pull request closed without merging**